appcelerator/titanium_mobile
# fix(android): Ti.App wrongly fires pause/resume events when opening/closing child windows

**JIRA:**
https://jira.appcelerator.org/browse/TIMOB-26746

**Summary:**
- Now only fires "pause" and "paused" event when app is backgrounded, like iOS.
- Now only fires "resume" and "resumed event when app is put into the foreground (including app startup), like iOS.
- Updated `Ti.App` docs to indicate Android supported these events since 7.5.0.
- We can't use Google's `ProcessLifecycleOwner` Java classes until the Titanium SDK can be built with gradle. This is because we need to use Google's lifecycle compiler JAR to handle its annotations. (Alternatively, we compile with Java 8, but then we could have a potential min Android OS version issue.)

**Test:**
1. Build and run the below code on Android.
2. Tap the "Open Child" button.
3. Observe log and verify that you do **NOT** see any pause/resume events.
4. Press the "Home" button to put app in the background.
5. Observe log and verify that you **do** see "pause" and "paused" events.
6. Resume the app.
7. Observe log and verify that you **do** see "resume" and "resumed" events.
8. Tap on the "Close Child" button.
9. Observe log and verify that you do **NOT** see any pause/resume events.

```javascript
function appEventHandler(e) {
	Ti.API.info("@@@ '" + e.type + "' event received.");
	if (Ti.App.Android || Ti.App.Windows) {
		Ti.UI.createNotification({
			message: e.type,
			duration: Ti.UI.NOTIFICATION_DURATION_SHORT,
		}).show();
	}
}

Ti.App.addEventListener("resume", appEventHandler);
Ti.App.addEventListener("resumed", appEventHandler);
Ti.App.addEventListener("pause", appEventHandler);
Ti.App.addEventListener("paused", appEventHandler);
Ti.App.addEventListener("close", appEventHandler);

var rootWindow = Ti.UI.createWindow({ title: "Parent Window" });
var openButton = Ti.UI.createButton({ title: "Open Child" });
openButton.addEventListener("click", function() {
	var childWindow = Ti.UI.createWindow({ title: "Child Window" });
	var closeButton = Ti.UI.createButton({ title: "Close Child" });
	closeButton.addEventListener("click", function() {
		childWindow.close();
	});
	childWindow.add(closeButton);
	childWindow.open();
});
rootWindow.add(openButton);
rootWindow.open();
```
